src/BuiltInTools/dotnet-watch/HotReload/CompilationHandler.cs

```diff
@@ -35,26 +34,22 @@ public CompilationHandler(IReporter reporter)
 
         public async ValueTask InitializeAsync(DotNetWatchContext context, CancellationToken cancellationToken)
         {
+            Debug.Assert(context.ProjectGraph is not null);
+
             if (_deltaApplier is null)
             {
-                _deltaApplier = context.DefaultLaunchSettingsProfile.HotReloadProfile switch
+                var hotReloadProfile = HotReloadProfileReader.InferHotReloadProfile(context.ProjectGraph, _reporter);
+                _deltaApplier = hotReloadProfile switch
                 {
-                    "blazorwasm" => new BlazorWebAssemblyDeltaApplier(_reporter),
-                    "blazorwasmhosted" => new BlazorWebAssemblyHostedDeltaApplier(_reporter),
-                    _ => new AspNetCoreDeltaApplier(_reporter),
+                    HotReloadProfile.BlazorWebAssembly => new BlazorWebAssemblyDeltaApplier(_reporter),
+                    HotReloadProfile.BlazorHosted => new BlazorWebAssemblyHostedDeltaApplier(_reporter),
+                    _ => new DefaultDeltaApplier(_reporter),
                 };
             }
 
             await _deltaApplier.InitializeAsync(context, cancellationToken);
 
-            if (context.Iteration == 0)
-            {
-                var instance = MSBuildLocator.QueryVisualStudioInstances().First();
-
-                _reporter.Verbose($"Using MSBuild at '{instance.MSBuildPath}' to load projects.");
-                MSBuildLocator.RegisterInstance(instance);
-            }
-            else if (_currentSolution is not null)
+            if (_currentSolution is not null)
```

Comment:
Why was this check removed?

Reply:
We register MSBuild super early in Program.Main and consequently no longer need to do it here.